File: TypeScript/CardOrganizer.ts
// i am not familir with TS so i used what more easy to me insted of enum class

export interface Card {
  name: string;
  date: string;
  leader: string;
}

export const CardOrganizer: Card[] = [
  { name: "BATTLE_OF_BADR", date: "2 AH", leader: "Prophet Muhammad" },
  { name: "BATTLE_OF_Mutta", date: "8 AH", leader: "Zayd ibn Harithah" },
  { name: "BATTLE_OF_QADISYYA", date: "15 AH", leader: "Saad ibn Abi Waqqas" },
  { name: "BATTLE_OF_YARMUK", date: "15 AH", leader: "Khalid ibn al-Walid" },
  { name: "BATTLE_OF_Nahavand", date: "21 AH", leader: "Al-Nuuman ibn Muqarrin" }
];

File: CardOrganizer.java
public enum CardOrganizer {
    // Constant varibles 
    BATTLE_OF_BADR("2 AH", "Prophet Muhammad"),
    BATTLE_OF_Mutta("8 AH", "Zayd ibn Harithah"),
    BATTLE_OF_QADISYYA("15 AH", "Saad ibn Abi Waqqas"),
    BATTLE_OF_YARMUK("15 AH", "Khalid ibn al-Walid"),
    BATTLE_OF_Nahavand("21 AH", "Al-Nuuman ibn Muqarrin");
    
    // I did it like slides in Lecture three
    private final String year;
    private final String leader;

    CardOrganizer(String year, String leader) {
        this.year = year;
        this.leader = leader;
    }

    public String getDate() {
        return year;
    }

    public String getLeader() {
        return leader;
    }
    
    
}
